### PR TITLE
Added migration code to populate unmigrated identity claims

### PR DESCRIPTION
**Pull request by sbryzak**
Migrates remaining users using their existing property values to populate `UserSignup.Spec.IdentityClaims`.

Fixes https://issues.redhat.com/browse/SANDBOX-463

e2e test: https://github.com/codeready-toolchain/toolchain-e2e/pull/838

### CODE REVIEW

File: controllers/usersignup/usersignup_controller.go [hunk 1]
```diff
@@ -106,6 +106,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	}
 	logger = logger.WithValues("username", userSignup.Spec.Username)
 
+	// TODO remove this section (and the referenced function) after migration has completed
+	// FROM HERE ---------
+	migrated, err := r.migrateUserSignupClaimsIfNecessary(ctx, userSignup)
```
Comment (alexeykazakov) [inline controllers/usersignup/usersignup_controller.go:111]: Let's move it after checking if the UserSignup resource is being deleted. There is no point of migrating the UserSignup which is being deleted.

**Review by alexeykazakov — COMMENTED** [review]
Looks good.

File: controllers/usersignup/usersignup_controller.go (hunk 1 above)
Comment (sbryzak) [reply controllers/usersignup/usersignup_controller.go:116]: Done.

**Review by alexeykazakov — COMMENTED** [review]
Looks good. But it's still missing unit tests for the migration step, right?

**Review by alexeykazakov — APPROVED** [review]

**Pull request merged**